### PR TITLE
Feat: add support for async tasks

### PR DESCRIPTION
**Pull request by thetutlage**
This PR adds support for `async tasks` as discussed in #29. 

## Summary

- [x] Updated README.
- [x] All tests are passing
- [x] Code linted

## Changes
1. Methods `run`, `runTask`, `runAlias` now returns `Promise`.
2. Synchronous errors from the above methods rejects the Promise, so that we have only one way of handle the error `.catch()`.
3. All of the tests now returns a `Promise`, since underlying methods are Promises.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -69,16 +75,25 @@ function run(name, directories, options, argv) {
...
+		if (!tasks) {
+			reject(new MrmUnknownAlias(`Alias “${aliasName}” not found.`));
+			return;
+		}
 
-	console.log(chalk.yellow(`Running alias ${aliasName}...`));
+		console.log(chalk.yellow(`Running alias ${aliasName}...`));
 
-	tasks.forEach(name => runTask(name, directories, options, argv));
+		tasks
```
Comment (sapegin) [inline src/index.js]: I’d extract this pattern into a function.

File: bin/mrm.js [hunk 2]
```diff
@@ -75,9 +75,7 @@ const options = getConfig(directories, 'config.json', argv);
 if (tasks.length === 0 || tasks[0] === 'help') {
 	commandHelp();
 } else {
-	try {
-		run(tasks, directories, options, argv);
-	} catch (err) {
+	run(tasks, directories, options, argv).catch(function(err) {
```
Comment (sapegin) [inline bin/mrm.js]: Let’s use an arrow function here ;-)

File: src/index.js [hunk 3]
```diff
@@ -42,6 +42,21 @@ function getAllAliases(options) {
...
+/**
+ * Runs an array of promises in series
+ *
+ * @method pSeries
+ *
+ * @param  {Array} items
+ * @param  {Function} iterator
+ * @return {Promise}
+ */
+function pSeries(items, iterator) {
```
Comment (sapegin) [inline src/index.js]: Let's call it `promiseSeries`, and we're done here ;-)

Comment (thetutlage) [reply src/index.js]: Sure!

**Pull request merged**